### PR TITLE
MES-6879 Migrate Common Cat D Test Report Components

### PR DESCRIPTION
**Pull request by Cullima**
## Description
- Migrates common test report components and related NGRX ready for full test report migration.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/test-report/components/end-test-modal/end-test-modal.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  </ion-row>
+  <ion-row>
+    <ion-col class="ion-no-padding column-border-top column-border-right">
+      <ion-button fill="clear" class="terminate-button" (click)="onTerminate()">Terminate test</ion-button>
+    </ion-col>
+    <ion-col class="ion-no-padding column-border-top">
+      <ion-button fill="clear" class="return-button" (click)="onCancel()">Return to test</ion-button>
+    </ion-col>
+  </ion-row>
+</ion-card>
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/end-test-modal/end-test-modal.html]: git doesnt like no EOF.

File: src/app/pages/test-report/components/eta-invalid-modal/eta-invalid-modal.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        At least one serious or dangerous fault should have been recorded.
+      </div>
+    </ion-col>
+  </ion-row>
+  <ion-row class="ion-justify-content-center">
+    <ion-col class="ion-no-padding" size="72">
+      <modal-return-button (onClick)="onCancel()">Return to test</modal-return-button>
+    </ion-col>
+  </ion-row>
+</ion-card>
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/eta-invalid-modal/eta-invalid-modal.html]: no newline EOF

File: src/app/pages/test-report/components/test-finalisation-invalid-test-data-modal/test-finalisation-invalid-test-data-modal.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  </ion-row>
+  <ion-row>
+    <ion-col class="column-border-top ion-no-padding">
+      <ion-button fill="clear" class="return-button" (click)="onCancel()">Change termination code</ion-button>
+    </ion-col>
+    <ion-col class="column-border-top ion-no-padding">
+      <ion-button fill="clear" class="terminate-button" (click)="onReturnToTestReport()">Return to test report</ion-button>
+    </ion-col>
+  </ion-row>
+</ion-card>
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/test-finalisation-invalid-test-data-modal/test-finalisation-invalid-test-data-modal.html]: another EOF

**Review by andrewsetterfield — APPROVED** [review]

File: src/app/pages/test-report/components/eco/eco.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+interface EcoComponentState {
+  completed$: Observable<boolean>;
+  adviceGivenPlanning$: Observable<boolean>;
+  adviceGivenControl$: Observable<boolean>;
+}
+
+@Component({
+  selector: 'eco',
+  templateUrl: 'eco.html',
```
Comment (tomcrawleyy) [inline src/app/pages/test-report/components/eco/eco.ts:24]: styleUrl missing

File: src/app/pages/test-report/components/reverse-diagram-link/reverse-diagram-link.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { REVERSE_DIAGRAM_PAGE } from '@pages/page-names.constants';
+import { AppComponent } from 'src/app/app.component';
+import {
+  ReverseDiagramClosed,
+  ReverseDiagramOpened,
+} from '../reverse-diagram-modal/reverse-diagram-modal.actions';
+
+@Component({
+  selector: 'reverse-diagram-link',
+  templateUrl: 'reverse-diagram-link.html',
```
Comment (tomcrawleyy) [inline src/app/pages/test-report/components/reverse-diagram-link/reverse-diagram-link.ts:20]: missing styleUrls here

File: src/app/pages/test-report/components/reverse-diagram-modal/__tests__/reverse-diagram-modal.spec.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
+/* eslint-disable */
```
Comment (tomcrawleyy) [inline src/app/pages/test-report/components/reverse-diagram-modal/__tests__/reverse-diagram-modal.spec.ts]: not sure we should be disabling eslint for entire pages

File: src/app/pages/test-report/components/serious-tooltip/serious-tooltip.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+
+// TODO: Serious Tooltip should not cover the Dangerous button
+@Component({
+  selector: 'serious-tooltip',
+  templateUrl: 'serious-tooltip.html',
```
Comment (tomcrawleyy) [inline src/app/pages/test-report/components/serious-tooltip/serious-tooltip.ts:6]: styleUrl missing

File: src/app/pages/test-report/components/reverse-left/reverse-left.actions.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { createAction } from '@ngrx/store';
+
+export const ReverseLeftPopoverOpened = createAction(
+  '[Reverse left] Reverse Left Popover Opened',
+);
+
+export const ReverseLeftPopoverClosed = createAction(
+  '[Reverse left] Reverse Left Popover Closed',
+);
```
Comment (tomcrawleyy) [inline src/app/pages/test-report/components/reverse-left/reverse-left.actions.ts:9]: there is no union function in this file, not sure if we are trying to be consistent with it or not

**Review by tomcrawleyy — CHANGES_REQUESTED** [review]
happy to approve once comments are addressed, looks good

File: src/app/pages/test-report/components/reverse-diagram-link/reverse-diagram-link.ts (hunk 5 above)
Comment (Cullima) [reply src/app/pages/test-report/components/reverse-diagram-link/reverse-diagram-link.ts:20]: added

File: src/app/pages/test-report/components/eco/eco.ts (hunk 4 above)
Comment (Cullima) [reply src/app/pages/test-report/components/eco/eco.ts:24]: added

File: src/app/pages/test-report/components/serious-tooltip/serious-tooltip.ts (hunk 7 above)
Comment (Cullima) [reply src/app/pages/test-report/components/serious-tooltip/serious-tooltip.ts:6]: added

File: src/app/pages/test-report/components/reverse-left/reverse-left.actions.ts (hunk 8 above)
Comment (Cullima) [reply src/app/pages/test-report/components/reverse-left/reverse-left.actions.ts:9]: only adding if they exist in previous app for now. As discussed, we can probably remove them all once all reducers are refactored

File: src/app/pages/test-report/components/test-finalisation-invalid-test-data-modal/test-finalisation-invalid-test-data-modal.html (hunk 3 above)
Comment (Cullima) [reply src/app/pages/test-report/components/test-finalisation-invalid-test-data-modal/test-finalisation-invalid-test-data-modal.html]: added

File: src/app/pages/test-report/components/eta-invalid-modal/eta-invalid-modal.html (hunk 2 above)
Comment (Cullima) [reply src/app/pages/test-report/components/eta-invalid-modal/eta-invalid-modal.html]: added

File: src/app/pages/test-report/components/end-test-modal/end-test-modal.html (hunk 1 above)
Comment (Cullima) [reply src/app/pages/test-report/components/end-test-modal/end-test-modal.html]: added

File: src/app/pages/test-report/components/reverse-diagram-modal/__tests__/reverse-diagram-modal.spec.ts (hunk 6 above)
Comment (Cullima) [reply src/app/pages/test-report/components/reverse-diagram-modal/__tests__/reverse-diagram-modal.spec.ts]: yeah good point have disabled just for spec that is throwing error. don't want to re-write at this point

**Review by tomcrawleyy — APPROVED** [review]

**Pull request merged**